src/vtkh/rendering/ScalarRenderer.cpp

```diff
@@ -137,23 +139,56 @@ ScalarRenderer::DoExecute()
   std::vector<std::string> field_names;
   PayloadCompositor compositor;
 
+  //Bounds needed for parallel execution
+  float bounds[6]; 
   for(int dom = 0; dom < num_domains; ++dom)
   {
-    Result res = renderers[dom].Render(m_camera);
-
-    field_names = res.ScalarNames;
-    PayloadImage *pimage = Convert(res);
-    min_p = std::min(min_p, pimage->m_payload_bytes);
-    max_p = std::max(max_p, pimage->m_payload_bytes);
-    compositor.AddImage(*pimage);
-    delete pimage;
+    vtkm::cont::DataSet data_set;
+    vtkm::Id domain_id;
+    m_input->GetDomain(dom, data_set, domain_id);
+    if(data_set.GetCellSet().GetNumberOfCells())
+    {
+	
+      Result res = renderers[dom].Render(m_camera);
+
+      field_names = res.ScalarNames;
+      PayloadImage *pimage = Convert(res);
+      min_p = std::min(min_p, pimage->m_payload_bytes);
+      max_p = std::max(max_p, pimage->m_payload_bytes);
+      compositor.AddImage(*pimage);
+      bounds[0] = pimage->m_bounds.X.Min;
+      bounds[1] = pimage->m_bounds.X.Max;
+      bounds[2] = pimage->m_bounds.Y.Min;
+      bounds[3] = pimage->m_bounds.Y.Max;
+      bounds[4] = pimage->m_bounds.Z.Min;
+      bounds[5] = pimage->m_bounds.Z.Max;
+      delete pimage;
+    }
+  }
+
+  //Assume rank 0 has data, pass details to ranks with empty domains (no data).
+#ifdef VTKH_PARALLEL
+  MPI_Bcast(bounds, 6, MPI_FLOAT, 0, MPI_COMM_WORLD);
+  MPI_Bcast(&max_p, 1, MPI_INT, 0, MPI_COMM_WORLD);
```

Comment:
Also, I am not sure that rank 0 has any promise of having any data. You will probably need to figure out who has data and make that the root rank.

Reply:
Can this be a future todo? I'm mainly trying to get some basic parallel runs done for the milestone. I'll probably have to fix this once I connect to exawind.

Reply:
Ok. I guess its no worse than what is was.